### ISSUE TITLE
[Question] Computed values are not allowed to cause side effects ?

### ISSUE THREAD

**Issue by mikecann** [mobxjs/mobx-state-tree#479]
I have the following:

```typescript
...
.views(self => ({
    
    get mergedMark() {
        if (self.selectedMarks.length < 2)
            return null;
        
        //return Mark.create(mergeMarks(self.selectedMarks.map(m => getSnapshot(m))), getEnv(self));
        return Mark.create({}, getEnv(self));
    },
    
}))
...
```

I want to return the result of a merge of multiple selected objects.

I get the following error:

>>> Uncaught Error: [mobx] Invariant failed: Computed values are not allowed to cause side effects by changing observables that are already being observed. Tried to modify: ObservableArray@1134

Im not sure I understand this error?

**Comment by mattiamanzati** [comment]
Yep, you cannot create or modify items in views, you could return a snapshot instead :)

**Comment by mikecann** [comment]
@mattiamanzati ah, I cant return an instance of the model in a view? I want to be able to call the actions on the model.

**Comment by mattiamanzati** [comment]
It's like performing "INSERT INTO" in a database sql view, makes no sense as it can create an endless loop :) Also would break time travelling as views have side effects.

What I can suggest you is to setup a reaction in the "afterCreate" lifecycle of the model, and update accordingly a prop with the value you are computing :)

**Comment by tinchoz49** [comment]
> What I can suggest you is to setup a reaction in the "afterCreate" lifecycle of the model, and update accordingly a prop with the value you are computing

cool pattern to include in the course of MST for eggheadio
